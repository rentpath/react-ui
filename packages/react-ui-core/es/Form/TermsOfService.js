import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

var _dec, _class, _class2, _temp;

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import themed from '@rentpath/react-themed';
import { Text } from '../Text';
var TermsOfService = (_dec = themed(/^TermsOfService/, {
  pure: true
}), _dec(_class = (_temp = _class2 =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(TermsOfService, _PureComponent);

  function TermsOfService() {
    _classCallCheck(this, TermsOfService);

    return _possibleConstructorReturn(this, _getPrototypeOf(TermsOfService).apply(this, arguments));
  }

  _createClass(TermsOfService, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          theme = _this$props.theme,
          className = _this$props.className,
          props = _objectWithoutProperties(_this$props, ["children", "theme", "className"]);

      return React.createElement(Text, _extends({
        className: classnames(className, theme.TermsOfService)
      }, props), children);
    }
  }]);

  return TermsOfService;
}(PureComponent), _defineProperty(_class2, "propTypes", {
  theme: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func])
}), _defineProperty(_class2, "defaultProps", {
  theme: {},
  children: null
}), _temp)) || _class);
export { TermsOfService as default };