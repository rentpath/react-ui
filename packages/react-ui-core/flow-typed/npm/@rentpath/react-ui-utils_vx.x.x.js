// flow-typed signature: a137d1bccb9d876def54c58cc5b6e8c1
// flow-typed version: <<STUB>>/@rentpath/react-ui-utils_v0.0.1/flow_v0.49.1

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@rentpath/react-ui-utils'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@rentpath/react-ui-utils' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@rentpath/react-ui-utils/lib/composeProps' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/index' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/keyMirror' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/parseArgs' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/randomId' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/composeProps-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/keyMirror-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/parseArgs-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/randomId-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/composeProps' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/index' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/keyMirror' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/parseArgs' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/randomId' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/test/setup' {
  declare module.exports: any;
}

// Filename aliases
declare module '@rentpath/react-ui-utils/lib/composeProps.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/composeProps'>;
}
declare module '@rentpath/react-ui-utils/lib/index.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/index'>;
}
declare module '@rentpath/react-ui-utils/lib/keyMirror.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/keyMirror'>;
}
declare module '@rentpath/react-ui-utils/lib/parseArgs.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/parseArgs'>;
}
declare module '@rentpath/react-ui-utils/lib/randomId.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/randomId'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/composeProps-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/composeProps-test'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/keyMirror-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/keyMirror-test'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/parseArgs-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/parseArgs-test'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/randomId-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/randomId-test'>;
}
declare module '@rentpath/react-ui-utils/src/composeProps.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/composeProps'>;
}
declare module '@rentpath/react-ui-utils/src/index.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/index'>;
}
declare module '@rentpath/react-ui-utils/src/keyMirror.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/keyMirror'>;
}
declare module '@rentpath/react-ui-utils/src/parseArgs.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/parseArgs'>;
}
declare module '@rentpath/react-ui-utils/src/randomId.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/randomId'>;
}
declare module '@rentpath/react-ui-utils/test/setup.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/test/setup'>;
}
