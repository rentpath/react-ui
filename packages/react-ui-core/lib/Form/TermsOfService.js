"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactThemed = _interopRequireDefault(require("@rentpath/react-themed"));

var _Text = require("../Text");

var _dec, _class, _class2, _temp;

var TermsOfService = (_dec = (0, _reactThemed.default)(/^TermsOfService/, {
  pure: true
}), _dec(_class = (_temp = _class2 =
/*#__PURE__*/
function (_PureComponent) {
  (0, _inherits2.default)(TermsOfService, _PureComponent);

  function TermsOfService() {
    (0, _classCallCheck2.default)(this, TermsOfService);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(TermsOfService).apply(this, arguments));
  }

  (0, _createClass2.default)(TermsOfService, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          theme = _this$props.theme,
          className = _this$props.className,
          props = (0, _objectWithoutProperties2.default)(_this$props, ["children", "theme", "className"]);
      return _react.default.createElement(_Text.Text, (0, _extends2.default)({
        className: (0, _classnames.default)(className, theme.TermsOfService)
      }, props), children);
    }
  }]);
  return TermsOfService;
}(_react.PureComponent), (0, _defineProperty2.default)(_class2, "propTypes", {
  theme: _propTypes.default.object,
  className: _propTypes.default.string,
  children: _propTypes.default.oneOfType([_propTypes.default.node, _propTypes.default.func])
}), (0, _defineProperty2.default)(_class2, "defaultProps", {
  theme: {},
  children: null
}), _temp)) || _class);
exports.default = TermsOfService;