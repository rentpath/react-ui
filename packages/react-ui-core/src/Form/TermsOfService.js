import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import themed from '@rentpath/react-themed'
import { Text } from '../Text'

@themed(/^TermsOfService/, { pure: true })
export default class TermsOfService extends PureComponent {
  static propTypes = {
    theme: PropTypes.object,
    className: PropTypes.string,
    children: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
    ]),
  }

  static defaultProps = {
    theme: {},
    children: null,
  }

  render() {
    const {
      children,
      theme,
      className,
      ...props
    } = this.props

    return (
      <Text
        className={clsx(
          className,
          theme.TermsOfService,
        )}
        {...props}
      >
        {children}
      </Text>
    )
  }
}
