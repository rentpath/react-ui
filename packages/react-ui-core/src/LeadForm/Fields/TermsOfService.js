import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import themed from 'react-themed'
import { Text } from '../../Text'

@themed(/^TermsOfService/, {
  pure: true,
})

export default class TermsOfService extends PureComponent {
  static propTypes = {
    theme: PropTypes.object,
    className: PropTypes.string,
    children: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
    ]),
  }

  static defaultProps = {
    theme: {},
    children: null,
  }

  render() {
    const {
      children,
      theme,
      className,
      ...props
    } = this.props

    return (
      <Text
        className={classnames(
          className,
          theme.TermsOfService,
        )}
        {...props}
      >
        {children}
      </Text>
    )
  }
}
