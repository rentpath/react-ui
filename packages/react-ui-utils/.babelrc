{
  "env": {
    "es": {
      "plugins": [
        "transform-decorators-legacy",
        "transform-class-properties",
        "babel-plugin-transform-object-rest-spread",
      ]
    },
    "cjs": {
      "presets": [
        "env"
      ],
      "plugins": [
        "transform-runtime",
        "transform-decorators-legacy",
        "transform-class-properties",
        "babel-plugin-transform-object-rest-spread"
      ]
    },
    "test": {
      "presets":[
        "env",
        "react"
      ],
      "plugins": [
        "transform-runtime",
        "transform-decorators-legacy",
        "transform-class-properties",
        "babel-plugin-transform-object-rest-spread"
      ]
    }
  }
}
