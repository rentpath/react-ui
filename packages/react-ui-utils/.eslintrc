{
  "parser": "babel-eslint",
  "extends": "airbnb-base",
  "rules": {
    "semi": [2, "never"],
    "arrow-parens": [2, "as-needed"],
    "no-underscore-dangle": [0],
    "import/no-extraneous-dependencies": ["error", {
      "devDependencies": true,
      "packageDir": "./package.json"
    }]
  },
  "settings": {
    "import/resolver": {
      "node": {
        "moduleDirectory": [
          "node_modules",
          "packages"
        ]
      }
    }
  },
  "env": {
    "mocha": true,
    "browser": true,
    "node": true
  }
}
