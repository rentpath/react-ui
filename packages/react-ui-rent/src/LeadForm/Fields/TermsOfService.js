import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import themed from '@rentpath/react-themed'
import { Text } from '@rentpath/react-ui-core'

@themed(/^TermsOfService/, {
  pure: true,
})

export default class TermsOfService extends PureComponent {
  static propTypes = {
    theme: PropTypes.object,
    className: PropTypes.string,
    children: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
    ]),
  }

  static defaultProps = {
    theme: {},
    children: (
      <div>
        {"By submitting this form, you agree to Rent.com's "}
        <a href="/company/legal/termsofservice/"> Terms of Service</a>
        {' and '}
        <a href="/company/privacy-full/"> Privacy Policy</a>
        .
      </div>
    ),
  }

  render() {
    const {
      children,
      theme,
      className,
      ...props
    } = this.props

    return (
      <Text
        className={classnames(
          className,
          theme.TermsOfService,
        )}
        {...props}
      >
        {children}
      </Text>
    )
  }
}
