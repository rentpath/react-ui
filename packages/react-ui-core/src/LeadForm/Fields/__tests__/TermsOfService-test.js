import React from 'react'
import renderer from 'react-test-renderer'
import { keyMirror } from '@rentpath/react-ui-utils'
import TermsOfService from '../TermsOfService'

const theme = keyMirror([
  'Text',
  'TermsOfService',
])

describe('TermsOfService', () => {
  it('renders default children if none passed', () => {
    const snap = renderer
      .create(<TermsOfService theme={theme} />)
      .toJSON()
    expect(snap).toMatchSnapshot()
  })

  it('renders children passed', () => {
    const snap = renderer
      .create(<TermsOfService theme={theme}>Foo</TermsOfService>)
      .toJSON()

    expect(snap).toMatchSnapshot()
  })
})
