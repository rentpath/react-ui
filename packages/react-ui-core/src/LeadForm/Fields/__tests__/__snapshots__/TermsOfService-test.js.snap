// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsOfService renders children passed 1`] = `
<div
  className="TermsOfService"
>
  Foo
</div>
`;

exports[`TermsOfService renders default children if none passed 1`] = `
<div
  className="TermsOfService"
/>
`;
